### PR TITLE
feat: MetaMask SDK integration

### PR DESCRIPTION
**Pull request by andreahaku**
Adds support for MetaMask SDK.
It also adds a "Disconnect" button to switch between MM extension and MM mobile wallet.

This version can be tested [here](https://c0f4f41c-2f55-4863-921b-sdk-docs.github.io/test-dapp/)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -366,11 +391,22 @@ const initialize = async () => {
...
       handleNewAccounts(newAccounts);
+      handleNewChain(ethereum.chainId);
     } catch (error) {
       console.error(error);
     }
   };
 
+  const onClickTerminate = async () => {
+    try {
+      sdk.terminate();
```
Comment (jiexi) [inline src/index.js:402]: does SDK termination restore `window.ethereum`?

**Comment by jiexi** [comment]
It seems connecting via extension via SDK and doesn't fully reconnect on page refresh. Clicking connect remembers the previous extension selection and uses it, but you still have to click connect again manually first

**Comment by andreahaku** [comment]
> It seems connecting via extension via SDK and doesn't fully reconnect on page refresh. Clicking connect remembers the previous extension selection and uses it, but you still have to click connect again manually first

@jiexi sorry I see your comment only now :) this is a draft PR yet and I'm waiting for the SDK to support EIP-6963 to make it a full PR and will take care of all those things as well. Thanks again

**Comment by jiexi** [comment]
hmm. i'm worried about if this will play nicely with https://github.com/MetaMask/test-dapp/pull/263

Is the goal to rely fully on MMSDK for establishing a connection to the provider?

**Comment by andreahaku** [comment]
no longer needed

**Pull request closed without merging**